functions/supabase/seed.sql

```diff
@@ -69,33 +74,75 @@ begin
     when 'contractComment' then 'comments'
     else null
   end;
+end
+$$;
+
+create or replace function replicate_writes_process_one(r incoming_writes)
+  returns boolean
+  language plpgsql
+as
+$$
+declare dest_table text;
+declare replicated_id text;
+begin
+  dest_table = get_document_table(r.doc_kind);
   if dest_table = null then
     raise warning 'Invalid document kind.';
-    return new;
+    return false;
   end if;
-  if new.write_kind = 'create' or new.write_kind = 'update' then
+  if r.write_kind = 'create' or r.write_kind = 'update' then
     execute format(
       'insert into %1$I (id, data, fs_updated_time) values (%2$L, %3$L, %4$L)
        on conflict (id) do update set data = %3$L, fs_updated_time = %4$L
        where %1$I.fs_updated_time <= %4$L
        returning id;',
-      dest_table, new.doc_id, new.data, new.ts
+      dest_table, r.doc_id, r.data, r.ts
     ) into replicated_id;
-  elsif new.write_kind = 'delete' then
+  elsif r.write_kind = 'delete' then
     execute format(
       'delete from %1$I where id = %2$L and fs_updated_time <= %3$L',
-      dest_table, new.doc_id, new.ts
+      dest_table, r.doc_id, r.ts
     ) into replicated_id;
   else
     raise warning 'Invalid write kind.';
-    return new;
+    return false;
   end if;
-  new.processed := true;
+  return true;
+end
+$$;
+
+create or replace function replicate_writes_process_all()
+  returns table(succeeded boolean, n bigint)
+  language plpgsql
+as
```

Comment:
This can be manually used to recover from the case where Firestore writes made it to the Supabase `incoming_writes` table, but failed to be replicated to the data tables. This will only happen if we wrote a bug (either in the trigger or in the cloud function populating `incoming_writes`) but we still need to be able to fix it when it happens.